File: bumperbot_cpp_examples/src/simple_service_client.cpp
#include <rclcpp/rclcpp.hpp>
#include <bumperbot_msgs/srv/add_two_ints.hpp>

using namespace std::chrono_literals;
using std::placeholders::_1;

class SimpleServiceClient: public rclcpp::Node
{
    public:
    SimpleServiceClient(int a, int b): Node("simple_service_client")
    {
        client_ = create_client<bumperbot_msgs::srv::AddTwoInts>("add_two_ints");
        while(!client_->wait_for_service(1s))
        {
            RCLCPP_INFO_STREAM(get_logger(), "Service not available, waiting again...");

        }

        
        req = std::make_shared<bumperbot_msgs::srv::AddTwoInts::Request>();
        req->a = a;
        req->b = b;

        auto future = client_->async_send_request(req, std::bind(&SimpleServiceClient::requestCallback, this, _1));

    }

    private:
        rclcpp::Client<bumperbot_msgs::srv::AddTwoInts>::SharedPtr client_;
        bumperbot_msgs::srv::AddTwoInts::Request::SharedPtr req;
        
        void requestCallback(rclcpp::Client<bumperbot_msgs::srv::AddTwoInts>::SharedFuture future)
        {
            if(future.valid())
            {
                RCLCPP_INFO_STREAM(get_logger(), "Service response " << future.get()->sum);
            }else{
                RCLCPP_ERROR(get_logger(), "Service Failure");
            }

        }
};


int main(int argc, char* argv[])
{
    int a = 8, b = 5;
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SimpleServiceClient>(a, b);
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: bumperbot_python_examples/bumperbot_python_examples/simple_service_client.py
import rclpy
from rclpy.node import Node
from bumperbot_msgs.srv import AddTwoInts
import sys

class SimpleServiceClient(Node):
    def __init__(self, a, b):
        super().__init__("simpe_service_client")
        self.client_ = self.create_client(AddTwoInts, "add_two_ints")

        while not self.client_.wait_for_service(timeout_sec=1.0):
            self.get_logger().info("Service not available, waiting again...")

        self.req = AddTwoInts.Request()
        self.req.a = a
        self.req.b = b

        self.future_ = self.client_.call_async(self.req)
        self.future_.add_done_callback(self.responseCallback)

    def responseCallback(self, future):
        self.get_logger().info(f"Service response {future.result().sum}")




def main():

    a = 6 
    b = 10

    rclpy.init()
    node = SimpleServiceClient(a, b)
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()


if __name__ == "__main__":
    main()